src/backend/commands/copy.c

```diff
@@ -4473,6 +4483,49 @@ CopyFrom(CopyState cstate)
 	return processed;
 }
 
+/*
+ * A subroutine to update the insert counts in all partitions for AO relations.
+ */
+static void CopyFromUpdateAOInsertCountInPartitions(HTAB *partition_hash, 
+													HTAB *tupcounts_ht,
+													List *ao_segnos)
+{
+	HASH_SEQ_STATUS hash_seq_status;
+	ResultPartHashEntry *entry;
+
+	hash_seq_init(&hash_seq_status, partition_hash);
+	while ((entry = hash_seq_search(&hash_seq_status)) != NULL)
+	{
+		ResultRelInfo *partInfo = &entry->resultRelInfo;
+
+		if (relstorage_is_ao(RelinfoGetStorage(partInfo)))
+		{
+			bool found;
+			Oid partRelid = RelationGetRelid(partInfo->ri_RelationDesc);
+			PQaoRelTupCount *ao = hash_search(tupcounts_ht, &partRelid,
+											  HASH_FIND, &found);
+			if (found)
+			{
+				ResultRelInfoSetSegno(partInfo, ao_segnos);
```

Comment:
Why do we need this line?

Reply:
The number of segments is already set [earlier](https://github.com/arenadata/gpdb/blob/fc2987c9cdf3210f602ce6799aeb085d0571157b/src/backend/commands/copy.c#L4119), so it seems like this line can be deleted.

However, the old code has a [similar call](https://github.com/arenadata/gpdb/blob/fc2987c9cdf3210f602ce6799aeb085d0571157b/src/backend/commands/copy.c#L4440) for the parent table, which has already been defined [earlier](https://github.com/arenadata/gpdb/blob/fc2987c9cdf3210f602ce6799aeb085d0571157b/src/backend/commands/copy.c#L3720).

So, I’m not entirely sure if these calls are really needed, and I couldn’t definitively determine it. In my tests, the behavior is identical whether the call is present or not, but I’m not certain that I’ve covered all possible cases. For that reason, I’d rather leave this call in place for now.